go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -1105,13 +1098,6 @@ func (h *defaultNetworkControllerEventHandler) DeleteResource(obj, cachedObj int
 		h.oc.eIPC.nodeZoneState.UnlockKey(node.Name)
 		return nil
 
-	case factory.EgressFwNodeType:
-		node, ok := obj.(*kapi.Node)
-		if !ok {
-			return fmt.Errorf("could not cast obj of type %T to *knet.Node", obj)
-		}
-		return h.oc.updateEgressFirewallForNode(node, nil)
```

Comment:
awesome!
this will a bug I have:
```
I0617 15:28:01.011366    9229 services_controller.go:519] Full service sync requested                                                                                        
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                      
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x1d6453a]                                                                                                     goroutine 109 [running]:
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*DefaultNetworkController).updateEgressFirewallForNode.func1({0x1f7d3c0?, 0xc000d9f230?}, {0x2174ae0?, 0xc001580700})
        /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/egressfirewall.go:797 +0x5fa
sync.(*Map).Range(0xc00080c168, 0xc001221a30)
        /usr/lib/golang/src/sync/map.go:476 +0x228
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*DefaultNetworkController).updateEgressFirewallForNode(0xc00080c000, 0x49a0d7?, 0x0)
        /go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/egressfirewall.go:776 +0x127
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*defaultNetworkC 
```
where we delete a node and we panic. This fix has to go all the way to 4.14... ^

Reply:
I am not gong to backport this fix, as it is quite a big change (and I specifically did a separate backportable part), but there is a very simple backportable fix for your bug, just add `newNode != nil` here https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/egressfirewall.go#L777

Reply:
added a unit test

Reply:
yeah thanks Nadia! yup I had thought of doing the node != nil fix but then wanted to check with you for backport plans first.
thanks for including that in your fix